x-pack/plugins/fleet/server/services/output.ts

```diff
@@ -549,8 +549,14 @@ class OutputService {
     esClient: ElasticsearchClient,
     id: string,
     data: Partial<Output>,
-    { fromPreconfiguration = false }: { fromPreconfiguration: boolean } = {
+    {
+      fromPreconfiguration = false,
+      // asHelper indicates that this update is being done by a helper function, and not by a user.
+      // This is used to prevent user from setting is_default or is_default_monitoring to false manually, leaving no default output.
+      asHelper = false,
```

Comment:
I'm not sure if it's the best design to add the `asHelper` param in the public service function.
Alternatively the `is_default` validation could remain in the public `update` function, and the rest of the update logic could be moved to a private function that would be called from both `create` and `update` functions.

Reply:
Agreed, much better idea than mine ;). Modified.